benchmark/CMakeLists.txt

```diff
@@ -2,80 +2,167 @@
 # Distributed under the Boost Software License, Version 1.0.
 # (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
 
+option(METABENCH_BRIGAND    "Benchmark Brigand" ON)
+option(METABENCH_FUSION     "Benchmark Fusion"  ON)
+option(METABENCH_HANA       "Benchmark Hana"    ON)
+option(METABENCH_META       "Benchmark Meta"    ON)
+option(METABENCH_METAL      "Benchmark Metal"   ON)
+option(METABENCH_MPL        "Benchmark MPL"     ON)
+
 include(ExternalProject)
 
-# Add several metaprogramming libraries that we want to benchmark
-find_package(Boost)
-if (Boost_FOUND)
-    include_directories(${Boost_INCLUDE_DIRS})
+if (WIN32)
+    set(BOOTSTRAP bootstrap)
+else()
+    set(BOOTSTRAP ./bootstrap.sh)
 endif()
 
-ExternalProject_Add(Brigand
-    GIT_REPOSITORY https://github.com/edouarda/brigand
-    GIT_TAG origin/master
-    TIMEOUT 10
-    CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-    PREFIX "${CMAKE_CURRENT_BINARY_DIR}"
-    BUILD_COMMAND ""   # Disable build step
-    INSTALL_COMMAND "" # Disable install step
-    TEST_COMMAND ""    # Disable test step
-)
-ExternalProject_Get_Property(Brigand SOURCE_DIR)
-include_directories(${SOURCE_DIR})
+if (WIN32)
+    set(B2 b2)
+else()
+    set(B2 ./b2)
+endif()
 
-ExternalProject_Add(Hana
-    GIT_REPOSITORY https://github.com/ldionne/hana
-    GIT_TAG origin/develop
-    TIMEOUT 10
-    CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-    PREFIX "${CMAKE_CURRENT_BINARY_DIR}"
-    BUILD_COMMAND ""   # Disable build step
-    INSTALL_COMMAND "" # Disable install step
-    TEST_COMMAND ""    # Disable test step
-)
-ExternalProject_Get_Property(Hana SOURCE_DIR)
-include_directories(${SOURCE_DIR}/include)
+if (METABENCH_FUSION OR METABENCH_MPL)
+    find_package(Boost QUIET)
+    if (Boost_FOUND)
+        message(STATUS "Boost found - version ${Boost_VERSION}")
+        include_directories(${Boost_INCLUDE_DIRS})
+        add_custom_target(Boost)
+    else()
+        find_package(Git QUIET)
+        if (GIT_FOUND)
+            message(STATUS "Boost not found - fetching branch master")
+            ExternalProject_Add(Boost
+                GIT_REPOSITORY https://github.com/boostorg/boost
+                GIT_TAG origin/master
+                GIT_SUBMODULES libs/config libs/core libs/detail libs/fusion libs/mpl
```

Comment:
I see two resolutions:

1. We do not install Boost by hand
2. We install Boost by hand, but we install __everything__, because I don't want the dependencies thing to blow up on me two years from now, when I don't remember any of this

I'm favouring (1), because installing all of Boost every time is going to be time consuming.